Add setStyles

excid3 opened the pull request:
This adds a `setStyles` operation that allows for setting multiple style attributes at once on an element. This actually loops through and delegates each entry to the `setStyle` operation.

For example, we can set background and color at once:

```ruby
cable_ready["MyChannel"].set_styles(
  selector: "string", # required - string containing a CSS selector or XPath expression
  styles: {
    background: "red",
    color: "white"
  }
)
```

Depending on browser compatibility targets / goals, we could shorten this by one line using `Object.entries` instead. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

hopsoft commented on javascript/cable_ready.js:
```diff
@@ -235,6 +235,15 @@ const DOMOperations = {
+      DOMOperations['setStyle']({ element: element, name: name, value: value })
+    }
+  },
+
```
Should we dispatch before/after events for this?

leastbad replied:
Won't those events be generated by the setStyle operations this creates?

leastbad replied:
Unless you see value in highlighting that several values were updated in addition to the events generated by the individual updates. I feel like it's a bit much, but I'm listening!

hopsoft replied:
It's definitely a lot of events, but it would be consistent. Also, wondering if we should suppress the individual events when `setSetyles` is used and only emit for the batch operation. I'm curious to see what @excid3 thinks.

excid3 replied:
Great questions. I feel like it'd be good to be consistent and have events for `setStyles`. 

We could also skip the delegation to `setStyle` if firing those individual events is a problem. 

I don't personally have a use for those events, so I'm not sure I've got the best opinion on those.

hopsoft commented:
I also like the idea of updating to use `Object.entries`

excid3 commented:
I'll update that to use Object.entries as well. 👍

excid3 commented:
I've updated this:

1. Using `Object.entries` now
2. Added events
3. Removed delegation to `setStyle` and just update styles in this operation itself

leastbad commented:
@excid3 I appreciated the cleverness of the initial solution, but agree that keeping them separated is likely the smart move. Looks good to me!
